### PR TITLE
fix(invity): electron receiver address

### LINKED ISSUE #2774: Invity - improve external pages opening in the desktop version
- move invity urls from allowedExternalUrls to invityUrls
- drop ELECTRON_RECEIVER_SERVER usage (buyUtils), always redirect to the same url regardless of the environment and handle request in electron handleExternalLink function (invityUrls validation)

### PR DESCRIPTION
**Pull request by vladimirvolek**
resolves: https://github.com/trezor/trezor-suite/issues/2774

### CODE REVIEW

File: packages/suite/src/utils/wallet/coinmarket/buyUtils.ts [hunk 1]
```diff
@@ -79,26 +78,28 @@ export function createQuoteLink(request: BuyTradeQuoteRequest, account: Account)
...
+};
 
-export function createTxLink(trade: BuyTrade, account: Account): string {
+export const createTxLink = async (trade: BuyTrade, account: Account) => {
     const locationOrigin = getLocationOrigin();
     const assetPrefix = process.env.assetPrefix || '';
     const params = `detail/${account.symbol}/${account.accountType}/${account.index}/${trade.paymentId}`;
     if (isDesktop()) {
-        return `${ELECTRON_RECEIVER_SERVER}/buy-redirect?p=${encodeURIComponent(
+        const serverUrl = await getElectronReceiverUrl();
```
Comment (szymonlesisz) [inline packages/suite/src/utils/wallet/coinmarket/buyUtils.ts]: the goal in this issue is to get rid of `if (desktop)` conditions from this file and use the same request url (line 101)
and do the "electron opening custom local server" logic on the electron side.

currently you've changed ELECTRON_RECEIVER_SERVER for getElectronReceiverUrl but this is not what we want here

Comment (vladimirvolek) [reply packages/suite/src/utils/wallet/coinmarket/buyUtils.ts]: But we are not handling the request, we are just passing it to the partner. We can't remove `if (desktop)` from here.

Comment (szymonlesisz) [reply packages/suite/src/utils/wallet/coinmarket/buyUtils.ts]: but that was the whole idea, to remove it. see issue description. discussed together with @martinboehm and you in person

File: packages/suite/src/utils/suite/env.ts [hunk 2]
```diff
@@ -48,13 +46,23 @@ export const getPlatformLanguage = () => {
 };
 
 export const isWeb = () => process.env.SUITE_TYPE === 'web';
+
 export const isDesktop = () => process.env.SUITE_TYPE === 'desktop';
 
+export const getElectronReceiverUrl = async () => {
+    if (isDesktop()) {
+        const addressData = await window.desktopApi!.getServerAddress();
```
Comment (szymonlesisz) [inline packages/suite/src/utils/suite/env.ts]:
```suggestion
export const getElectronReceiverUrl = (pathname: '/buy-post' | '/buy-redirect') => {
   return window.desktopApi?.getHttpReceiverAddress(pathname);
```
looks like you don't have to use (and export) `getServerAddress`

`getHttpReceiverAddress` is already there and do exactly what you want

**Review by szymonlesisz — CHANGES_REQUESTED** [review]

File: packages/suite/src/hooks/wallet/useCoinmarketBuyOffers.ts [hunk 3]
```diff
@@ -97,7 +97,7 @@ export const useOffers = (props: Props) => {
                 if (!quote.quoteId) {
                     const response = await invityAPI.doBuyTrade({
                         trade: quote,
-                        returnUrl: createQuoteLink(quotesRequest, account),
+                        returnUrl: await createQuoteLink(quotesRequest, account),
```
Comment (szymonlesisz) [inline packages/suite/src/hooks/wallet/useCoinmarketBuyOffers.ts]: declare `returnUrl` as a const before the invityAPI call.

async in async looks suspicious

File: packages/suite/src/hooks/wallet/useCoinmarketBuyOffers.ts [hunk 4]
```diff
@@ -127,7 +127,7 @@ export const useOffers = (props: Props) => {
         const quote = { ...selectedQuote, receiveAddress: address };
         const response = await invityAPI.doBuyTrade({
             trade: quote,
-            returnUrl: createTxLink(selectedQuote, account),
+            returnUrl: await createTxLink(selectedQuote, account),
```
Comment (szymonlesisz) [inline packages/suite/src/hooks/wallet/useCoinmarketBuyOffers.ts]: declare `returnUrl` as a const before the invityAPI call.

async in async looks suspicious

File: packages/suite/src/views/wallet/coinmarket/buy/detail/components/WaitingForPayment/index.tsx [hunk 5]
```diff
@@ -58,12 +58,12 @@ const WaitingForPayment = ({ transactionId, trade, account }: Props) => {
...
     });
 
-    const goToPayment = () => {
+    const goToPayment = async () => {
         setIsWorking(true);
         invityAPI
             .getBuyTradeForm({
                 trade,
-                returnUrl: createTxLink(trade, account),
+                returnUrl: await createTxLink(trade, account),
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/coinmarket/buy/detail/components/WaitingForPayment/index.tsx]: declare `returnUrl` as a const before the invityAPI call.

**Review by szymonlesisz — CHANGES_REQUESTED** [review]

**Review by szymonlesisz — APPROVED** [review]
squash your commits and rename it to something more accurate, like: `fix(invity): electron receiver address`

**Pull request merged**